tools/Repo-Tasks.psm1

```diff
@@ -0,0 +1,253 @@
+﻿$taskScriptDir = [System.IO.Path]::GetDirectoryName($PSCommandPath)
+$env:repoRoot = [System.IO.Path]::GetDirectoryName($taskScriptDir)
+
+#$userPsFileEnvVariable = $env:psuserpreferences
+$userPsFileDir = [string]::Empty
+
+[string]$envVariableName="TEST_CSM_ORGID_AUTHENTICATION"
+
+<#
+We allow users to include any helper powershell scripts they would like to include in the current session
+Currently we support two ways to include helper powershell scripts
+1) psuserspreferences environment variable
+2) $env:USERPROFILE\psFiles directory
+We will include all *.ps1 files from any of the above mentioned locations
+#>
+if([System.IO.Directory]::Exists($env:psuserpreferences))
+{
+    $userPsFileDir = $env:psuserpreferences
+}
+elseif([System.IO.Directory]::Exists("$env:USERPROFILE\psFiles"))
+{
+    $userPsFileDir = "$env:USERPROFILE\psFiles"
+}
+
+if([string]::IsNullOrEmpty($userPsFileDir) -eq $false)
+{
+    Get-ChildItem $userPsFileDir | WHERE {$_.Name -like "*.ps1"} | ForEach {
+    Write-Host "Including $_" -ForegroundColor Green
+    . $userPsFileDir\$_
+    }
+}
+else
+{
+    Write-Host "Loading skipped. 'psuserpreferences' environment variable was not set to load user preferences." -ForegroundColor DarkYellow
+}
+
+[CmdletBinding]
+Function Set-TestEnvironment
+{
+<#
+.SYNOPSIS
+This cmdlet helps you to setup Test Environment for running tests
+In order to successfully run a test, you will need SubscriptionId, TenantId
+This cmdlet will only prompt you for Subscription and Tenant information, rest all other parameters are optional
+
+#>
+    param(
+        [parameter(Mandatory=$true, Position=0, HelpMessage='SubscriptionId you would like to use')]
+        [ValidateNotNullOrEmpty()]
+        [string]$SubscriptionId,
+
+        [parameter(Mandatory=$true, Position=1, HelpMessage='AADTenant/TenantId you would like to use:')]
```

Comment:
Here are the parameter sets:

(1) Service Principal With Secret (TenantId = required (Position 1), PSCredential ServicePrincipalCredentials = required, [Position = 2])
(2) UserID [default parameter set] PSCredential UserCredential = optional

The following parameters belong to all parameter sets:
SubscriptionId=required
Environment=optional ValidateSet=Prod| Dogfood| Current | Next, Default=Prod)
RecordMode=optional, ValidateSet=(Record, None, Playback, default = Playback)